deps/fast_float_c_interface/Makefile

```diff
@@ -0,0 +1,15 @@
+CXX?=c++
+# FASTFLOAT_ALLOWS_LEADING_PLUS allows +inf to be parsed as inf, instead of error. 
+CXXFLAGS=-std=c++11 -fPIC -D FASTFLOAT_ALLOWS_LEADING_PLUS
+
+.PHONY: all clean
+all: valkey_strtod.o
+clean:
+	rm -f *.o || true;
+
+# src/Makefile passes values for OPT, STD and WARN variables. These are core CFLAGS used by executable targets.
+# We don't need to pass USE_FAST_FLOAT value to compiler here. The code in src/ includes the header files as is,
+# and at that point disables fast_float code based on USE_FAST_FLOAT value. Therefore, only src/Makefile passes 
+# USE_FAST_FLOAT value to compiler.
+valkey_strtod.o: valkey_strtod.cpp valkey_strtod.h
+	$(CXX) $(OPT) $(STD) $(WARN) $(CXXFLAGS) -o $@ -c $<
```

Comment:
The `deps/Makefile` passes the CFLAGS variable, but not the OPT, STD and WARN flags, so these are not defined here. That was probably a typo in deps/Makefile.

The STD passed from src/Makefile contains `-std=c11`, which is not appropriate for C++. The fast float code is written for C++11 so we should use that for fast float.

The WARN we use in src/Makefile contains some warning that is invalid in C++. I got a warning locally when I tried to use the same flags for C++ that we use for C. Each project uses differnt code style and needs a different set of warnings, so let's use whatever fast float recommends, or something generic like `-Wall -Wextra`.

Maybe we can pass OPT and DEBUG from src/Makefile.

Reply:
I cleaned up all of this.